### PR TITLE
Add zipN method

### PR DESCRIPTION
**Pull request by kborucinski**
Add zipN method which zips given arrays

### CODE REVIEW

File: array/zipN.ts [hunk 1]
```diff
@@ -0,0 +1,2 @@
+export default (xs: any[], ...ys: any[]) =>
```
Comment (przemyslawzalewski) [inline array/zipN.ts]: Please make the interface be `(...xs: any[]): any[]`

File: array/zipN.test.ts [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+
+  it("should zips three arrays with the same lengths", () => {
+    expect(zipN([1, 2, 3], [4, 5, 6], [7, 8, 9])).toEqual([
+      [1, 4, 7],
+      [2, 5, 8],
+      [3, 6, 9]
+    ]);
+  });
+
+  it("should zips three arrays with different lengths", () => {
```
Comment (przemyslawzalewski) [inline array/zipN.test.ts]: Please add a test case for a single element and an empty call.

File: array/zipN.test.ts [hunk 3]
```diff
@@ -0,0 +1,57 @@
+/* eslint-env jest */
+// @ts-ignore ambiguous import
+import zipN from "./zipN.ts";
+
+describe("zip", () => {
+  it("should zips with pair constructor", () => {
```
Comment (marekrozmus) [inline array/zipN.test.ts]: zips => zip ?

File: array/zipN.js [hunk 4]
```diff
@@ -1,2 +1,7 @@
-export default (xs, ...ys) =>
-  xs.map((value, index) => ys.reduce((x, xs) => [...x, xs[index]], [value]));
+export default (...xs) => {
+  const [head, ...tail] = xs;
+
+  return head.map((value, index) =>
```
Comment (przemyslawzalewski) [inline array/zipN.js]: This will fail for the case `zipN()` - please guard against it and cover it with tests.

File: array/zipN.ts [hunk 5]
```diff
@@ -0,0 +1,7 @@
+export default (...xs: any[]) => {
```
Comment (przemyslawzalewski) [inline array/zipN.ts]: The type should be `any[][]`.
